src/queue.js

```diff
@@ -219,4 +219,6 @@ class Queue extends EventEmitter {
   }
 }
 
+Queue.Job = null; // Make TypeScript compiler happy. Needed because Job is an abstract class.
```

Comment:
Ah, awesome explanation @viridia thanks.

When I designed the Job object I first had a data property. Later I realized how flexible it could be to allow the user to add any property they like (the power of JS objects).   I still believe the decision to allow full customization of the jobs to be a good one. That said I can appreciate how difficult it would be for a Type system to work with the Job objects, hence the abstract class.

OK, so the cost of having a flexible Job object is one extra property on the Queue object. I can swallow that to support TypeScript.

Now, is having `Queue.Job` equal to `null` the best option? Also does the `Queue.Job` need to have a capital `J`?